### PR TITLE
[SLP][REVEC] Fix cost model for getGatherCost with FixedVectorType ScalarTy.

### PR DESCRIPTION
**Pull request by HanKuanChen**

### CODE REVIEW

File: llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp [hunk 1]
```diff
@@ -11767,14 +11765,27 @@ InstructionCost BoUpSLP::getGatherCost(ArrayRef<Value *> VL, bool ForPoisonSrc,
...
-        TTI->getScalarizationOverhead(VecTy, ~ShuffledElements, /*Insert*/ true,
-                                      /*Extract*/ false, CostKind);
+  if (ForPoisonSrc) {
+    if (isa<FixedVectorType>(ScalarTy)) {
+      assert(SLPReVec && "Only supported by REVEC.");
+      // We don't need to insert elements one by one. Instead, we can insert the
+      // entire vector into the destination.
+      Cost = 0;
+      unsigned ScalarTyNumElements = getNumElements(ScalarTy);
+      for (unsigned I = 0, E = VL.size(); I != E; ++I)
```
Comment (alexey-bataev) [inline llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp:11778]:
```suggestion
      for (unsigned I : seq<unsigned>(VL.size()))
```

File: llvm/test/Transforms/SLPVectorizer/RISCV/revec-getGatherCost.ll [hunk 2]
```diff
@@ -0,0 +1,31 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py
+; RUN: opt -mtriple=riscv64 -mcpu=sifive-x280 -passes=slp-vectorizer -S -slp-revec -slp-max-reg-size=1024 -slp-threshold=-10 %s | FileCheck %s
```
Comment (alexey-bataev) [inline llvm/test/Transforms/SLPVectorizer/RISCV/revec-getGatherCost.ll:2]: Would be good to add remarks check to see cost changes

Comment (HanKuanChen) [reply llvm/test/Transforms/SLPVectorizer/RISCV/revec-getGatherCost.ll]: done

**Review by alexey-bataev — APPROVED** [review]
LG

**Pull request merged**